### PR TITLE
Upcoming Activities Page [Papilio 28]

### PR DESCRIPTION
**Pull request by DrakeHunterK7**
# General info

**Issue number**: PAPILIO-28

**Task description**: 

 - Added ability to view all the user's created and joined activities all on one page, the upcoming activities page

# Testing
To test this feature you have to:

1. Log in or sign up to the app
2. Create an activity. Note the date of this activity. 
3. Go to the "My Activities" tab and the first thing you'll see is the upcoming activities page. Wait for the upcoming activities to load. 
4. Once loading is done, you will see a chronological view of activities ordered by the date (from now to end of December). Look for the activity you created here. 
5. Verify that the activity you created appears on this page under the correct month. Also verify that in the activity card, this symbol is shown:

image

6. Also verify that you can tap on the card and get taken to that activity's details page. 
7. Now go back to the Activity Feed page and join any activity that you have not created. 
8. Now go to the Upcoming Activities page again and wait for it to load again
9. Verify that the activity you recently joined is shown here under the correct month. Also verify that the activity card does NOT show the symbol above. 
10. Verify that tapping on the month container of the activity hides/shows activities for that month.  
11. If steps 5, 6, 9, and 10 are verified in the affirmative, then this story is complete. 

# Notes
- UI subject to improvements later on.

### CODE REVIEW

File: app/src/main/java/com/soen490chrysalis/papilio/repository/users/IUserRepository.kt [hunk 1]
```diff
@@ -43,6 +40,11 @@ interface IUserRepository
 
     suspend fun getFavoriteActivities() : Response<FavoriteActivitiesResponse>
 
+    suspend fun getJoinedActivities() : Response<JoinedActivitiesResponse>
+
+    suspend fun getCreatedActivities() : Response<FavoriteActivitiesResponse>
+
+
```
Comment (AC159) [inline app/src/main/java/com/soen490chrysalis/papilio/repository/users/IUserRepository.kt:47]: Remove this extra line

File: app/src/main/java/com/soen490chrysalis/papilio/viewModel/UpcomingActivitiesViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+class UpcomingActivitiesViewModel(private val userRepository: IUserRepository) : ViewModel() {
+
+    private val logTag = UpcomingActivitiesViewModel::class.java.simpleName
+    var activitiesResponse: MutableLiveData<FavoriteActivitiesResponse> =
+        MutableLiveData<FavoriteActivitiesResponse>()
+
+    fun getUpcomingActivities() {
+        viewModelScope.launch {
+
+              try {
```
Comment (AC159) [inline app/src/main/java/com/soen490chrysalis/papilio/viewModel/UpcomingActivitiesViewModel.kt:23]: Apply formatting

File: app/src/main/res/layout/fragment_activities.xml [hunk 3]
```diff
@@ -1,21 +1,88 @@
...
-        android:foregroundTint="@color/white"
-        android:backgroundTint="@color/dark_blue"
-        android:layout_margin="16dp"
-        android:contentDescription="@string/create_activity" />
+    <LinearLayout
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:orientation="vertical">
+
+        <ImageView
```
Comment (AC159) [inline app/src/main/res/layout/fragment_activities.xml:15]: Placing our logo here does not make sense to me, screen space is already limited on phones and we add the logo here to block a quarter of the screen! I would just remove it.
I would also place the "create activity" button at the bottom of the screen to make it easily reachable

Comment (DrakeHunterK7) [reply app/src/main/res/layout/fragment_activities.xml:15]: You got it

**Review by AC159 — APPROVED** [review]
Good to go

**Pull request merged**